### ISSUE TITLE
Continue implementation for conditional connections between rooms and testing

### ISSUE THREAD

**Issue by victoriavillalba** [uchicago-cs/chiventure#575]
Wiki for plan: 
https://github.com/uchicago-cs/chiventure/wiki/Conditional-Connections-Between-Rooms-Design-Plan.

Continue issue from Sprint 2, issue #514. Code will be implemented room.c, path.c, item.c, and other files in game-state. The code will allow rooms to have pre-requisites that must be completed before being able to enter the room.

Goals include:

Talking to game-state/wdl/etc. groups
Finish implementing code
Finish running tests
Push code

**Comment by victoriavillalba** [comment]
Added char* trigger value to action_type_t struct, linked list answers value to path_t struct, and found that some of the path functions have not been implemented, which should be done by us, but will talk to game-state group.

**Comment by victoriavillalba** [comment]
Also added the char* trigger parameter to path_new function in path.h, which is one of the functions that haven't been implemented yet.

**Comment by victoriavillalba** [comment]
The allow_path_access function that we thought of using can actually be replaced with get/set_attribute functions in the item.c file so we now know that that one won't be necessary.

**Comment by victoriavillalba** [comment]
We found that path.c functions have their headers in room.h, so we should consider moving those to path.h

**Comment by victoriavillalba** [comment]
New problems found in path.c: delete_all_conditions function will need to be made for path_free and path_free needs to free items associated with path.

**Comment by victoriavillalba** [comment]
I had been having problems pushing and turns out I misspelled the name of the branch and now there's an accidental duplicate of the branch in chi venture, but I'm going to delete it.

**Comment by victoriavillalba** [comment]
Deleted accidental branch

**Comment by victoriavillalba** [comment]
Concern that just popped up: how do we go to the room struct using only the room_id

**Comment by victoriavillalba** [comment]
Main function has been written, but the helper function needs to be written and there are no tests yet. Function is supposed to remove action from list of conditions when called. The function will then have to be added to the main action functions in action management.c.

**Comment by victoriavillalba** [comment]
Updated old tests and functions to support our additional parameters structs, made a test for the new function in action management.c, but haven't run the tests yet to see if it all works because I don't know how yet. Realized that we need to make a few more functions to support having a list of action types in path_t:
function that adds nodes in linked list
update path tests
update path new/free/init

and then a new test file needs to be written for the new function in path.c/room.h

**Comment by victoriavillalba** [comment]
Last things to do:
1. write test for remove_condition()
2. Implement remove_condition() in actionmanagement.c do functions
3. Test to see if everything actually works

**Comment by victoriavillalba** [comment]
Tests were made and run, but the tests were failing so we will continue the next sprint with fixing the tests and then making the demo for the hopefully functional conditional rooms.

**Comment by MaxineK36** [comment]
Issue Score: ✔️+

Comments:
Good work! Two things to work on for future issues:
1. The title of an issue has to convey what the issue will be producing. In the future, try using simple titles with a "Verb-Object" structure that clearly states what you will be doing in the issue (e.g., “Design and implement a ‘circle’ module”, “Research libraries for drawing simple geometric shapes”, etc.). "Continue implementation" doesn't convey a clear goal, since it's unclear when it would be done. 
2. The issue was closed before it was actually done; instead of being closed, it should have been pushed over to the next sprint.
